exp-models/addon/models/experiment.js

```diff
@@ -29,6 +29,40 @@ export default DS.Model.extend(JamModel, {
     // A flag for whether or not the participant has seen this feedback
     hasReadFeedback: DS.attr('boolean'),
 
+    // This needs to be a seperate collection because string fields of a certain length
+    // cannot be indexed by Elasticsearch.
+    thumbnailId: DS.attr('string'),
+    thumbnail: Ember.computed('thumbnailId', {
+        get() {
+            var thumbnailId = this.get('thumbnailId');
+            if (thumbnailId) {
+                return this.get('store').findRecord('thumbnail', thumbnailId);
+            }
+            return null;
+        },
+        set (raw) {
+            var self = this;
+
+            var thumbnailId = this.get('thumbnailId');
+            if (thumbnailId) {
```

Comment:
Using computed getters and setters is a nice idea. :) Just to go further- can the setter call the getter value?

(specifically, I'm noticing that the store.findRecord code is repeated)

Not a huge issue if it works.